libs/src/services/hedgehog/index.js

```diff
@@ -86,6 +86,48 @@ class HedgehogWrapper {
       }
     }
 
+    /**
+     * Generate secure credentials to allow login
+     * @param {String} username username
+     */
+    hedgehog.getEntropyFromLocalStorage = async () => {
+      let entropy = await WalletManager.getEntropyFromLocalStorage()
+      if (entropy == null) {
+        throw new Error('GenerateRecoveryInfo | Entropy missing')
+      }
+      return entropy
+    }
+
+    /**
+     * Generate new set of auth credentials to allow login
+     * @param {String} username - username
+     * @param {String} password - new password
+     * @param {String} entropy - stored entropy val
+     */
+    hedgehog.resetPassword = async (username, password, entropy) => {
+      let self = this
```

Comment:
I think we can stop this use of self to refer to this. This is kind of an old pattern which is somewhat out of date now

Reply:
is this safe to do? should i just remove it elsewhere too? btw this is getting moved to hedgehog module anyway

Reply:
as long as you remove references to `self.` in this function it's all good. and yeah we should definitely move this into hedgehog